src/main/java/mk/route/routemk/services/user/TicketServiceImpl.java

```diff
@@ -23,4 +26,15 @@ public TicketServiceImpl(GenericRepository<Ticket, Integer> genericRepository, T
     public Collection<? extends Ticket> findTripsByAccountId(Integer accountId) {
         return ticketRepository.findTicketsByAccount_AccountId(accountId);
     }
+
+    @Override
+    public Ticket findCheapestTicketForTrip(Integer tripId) {
+        List<Ticket> tickets =  ticketRepository.findAllByTrip_TripId(tripId).
+                stream().sorted(Comparator.comparing(Ticket::getPrice)).collect(Collectors.toList());
+
+        if (tickets.isEmpty()) {
+            return null; // or throw an exception
+        }
+        return tickets.get(0); // return the first ticket, the cheapest due to sorting
```

Comment:
you can one shot this function using the `min` stream function
```suggestion
       return ticketRepository.findAllByTrip_TripId(tripId)
                   .stream()
                   .min(Comparator.comparing(Ticket::getPrice))
                   .orElse(null);
```